app/layersmodel.cpp

```diff
@@ -63,10 +63,31 @@ void LayersModel::reloadLayers()
     mLayers = allLayers;
     endResetModel();
 
-    emit layersChanged();
+    emit layersReloaded();
   }
 }
 
+void LayersModel::updateActiveLayer( const QString &name )
+{
+  int row = rowAccordingName( name, firstNonOnlyReadableLayerIndex() );
+  setActiveIndex( row );
+}
+
+int LayersModel::activeIndex() const
+{
+  return mActiveIndex;
+}
+
+void LayersModel::setActiveIndex( int activeIndex )
+{
+  if ( mActiveIndex != activeIndex )
+  {
+    mActiveIndex = activeIndex;
+  }
+  emit activeIndexChanged();
```

Comment:
should we emit only when index is actually changed?

Reply:
Signal is used in main.qml to update the RecordToolbar. Consider a use case when mapTheme has changed, layers in the layers model have changed, (default/active) index has not changed, but actual layer where the index is pointing at is different from the previous state.

A new signal can be introduced and used for updating the RecordToolbar instead + added docs.

Reply:
i would emit only when index is changed, and for the testcase you describe I imaging there is signal in maptheme that is emitted already?